implementations/rust/ockam/ockam_command/src/run/parser.rs

```diff
@@ -0,0 +1,323 @@
+                    "--to",
+                    &inlet.to,
+                    "--alias",
+                    name,
+                ];
+                insert_command("inlet", name, None, args)?;
+            }
+        }
+
+        if let Some(tcp_outlets) = &self.tcp_outlets {
+            for (name, outlet) in tcp_outlets {
+                // TODO: store outlets in CliState; Then check if the outlet already exists. If it doesn't, create it.
+                if let Some(exp) = &outlet.access_control {
+                    let args = &[
+                        "policy",
+                        "set",
+                        "--at",
+                        &node_name_formatted,
+                        "--resource",
+                        "tcp-outlet",
+                        "--expression",
+                        exp,
+                    ];
+                    insert_command("policy", name, None, args)?;
+                }
+                let args = &[
+                    "tcp-outlet",
+                    "create",
+                    "--at",
+                    &node_name_formatted,
+                    "--from",
+                    &outlet.from,
+                    "--to",
+                    &outlet.to,
+                    "--alias",
+                    name,
+                ];
+                insert_command("outlet", name, None, args)?;
+            }
+        }
+
+        if let Some(forwarders) = &self.forwarders {
+            for (name, forwarder) in forwarders {
+                // TODO: store forwarders in CliState; Then check if the forwarder already exists. If it doesn't, create it.
+                let args = &[
+                    "forwarder",
+                    "create",
+                    name,
+                    "--to",
+                    &node_name_formatted,
+                    "--at",
+                    &forwarder.at,
+                ];
+                insert_command("forwarder", name, None, args)?;
+            }
+        }
+
+        Ok(())
+    }
+}
+
+/// Defines the structure of a tcp inlet in the config file.
+#[derive(Debug, Deserialize)]
+pub struct InletConfig {
+    pub from: String,
+    pub to: String,
+    pub access_control: Option<String>,
+}
+
+/// Defines the structure of a tcp outlet in the config file.
+#[derive(Debug, Deserialize)]
+pub struct OutletConfig {
+    pub from: String,
+    pub to: String,
+    pub access_control: Option<String>,
+}
+
+/// Defines the structure of a forwarder in the config file.
+#[derive(Debug, Deserialize)]
+pub struct ForwarderConfig {
+    pub at: String,
+}
+
+static BINARY_PATH: Lazy<String> = Lazy::new(|| {
+    if let Ok(p) = std::env::var("OCKAM") {
+        p
+    } else {
+        "ockam".to_string()
+    }
+});
+
+fn binary_path() -> &'static str {
+    &BINARY_PATH
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_parse_config_with_depends_on() {
+        let config = r#"
+            nodes:
```

Comment:
I can see HCL as another suitable candidate for commands representation

Reply:
The current structure is going to change in the next iteration, but as soon as we have that part more stabilized, we could add support to any format we want I think.